### PR TITLE
Library repository unit tests

### PR DESCRIPTION
**Pull request by calebschoepp**
Depends on #169 

# Summary
- Unit tests for Library repository
- Couldn't get test the methods that used Firestore transactions. I think there is a bug where transactions don't properly work in the Firestore emulator
- I don't believe there is any flakiness in these tests. That being said it would be ideal if we could flush the DB after every test. https://firebase.google.com/docs/emulator-suite/connect_firestore#clear_your_database_between_tests

Closes insert-issue-number-here

### CODE REVIEW

File: app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java [hunk 1]
```diff
@@ -67,11 +79,17 @@ public void setup() {
...
+    public void cleanup() throws IOException {
+        mDb.terminate();
+    }
+
     @Test
     public void addDeleteBookTest() throws InterruptedException {
         Book book = new Book("9780441016075", "Halting State", "Charles Stross", mAuth.getCurrentUser().getUid(), null);
 
-        mRepository.createBook(book, documentReference -> {}, e -> fail("Unable to create book: " + e.getMessage()));
+        mRepository.createBook(book, documentReference -> {
```
Comment (cyrusdiego) [inline app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java:91]: nit: revert

Comment (cyrusdiego) [reply app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java:91]: oh wait i noticed line 118 is like this as well, I'm assuming it's the formatted that did this, can just leave it

File: app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java [hunk 2]
```diff
@@ -109,4 +128,269 @@ public void addDeleteBookTest() throws InterruptedException {
...
+
+        mRepository.deleteBook(book, aVoid -> {
+        }, e -> fail("Unable to delete book " + e.getMessage()));
+
+        Thread.sleep(SLEEP_TIME_MILLIS);
+        shadowOf(getMainLooper()).idle();
+    }
+
+//    @Test
+//    public void declineRequester() throws InterruptedException {
```
Comment (cyrusdiego) [inline app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java:326]: Was there an issue / was it too difficult to test this?

**Review by cyrusdiego — APPROVED** [review]
LGTM! I just left some comments as well but minor

File: app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java (hunk 2 above)
Comment (calebschoepp) [reply app/src/test/java/com/example/unlibrary/library/LibraryRepositoryTest.java:326]: This was the code that used transactions that I could not get to work on the emulator.

**Pull request merged**